src/Tec/Plugin.php

```diff
@@ -104,26 +139,935 @@ public function register() {
+						'ping_status'    => 'closed',
+					];
+					$success = wp_update_post( $args );
+
+					// Logging
+					$msg = "Updating post name and post parent for Attendee ";
+					$msg .= $success ? "successful" : "NOT successful";
+					$this->add_to_log( $msg );
+				}
+			}
+		}
+	}
+
+	/**
+	 * Hashing function.
+	 *
+	 * @param string $subject The subject to be hashed.
+	 *
+	 * @return string The hashed string.
+	 */
+	function hashit( string $subject ): string {
+		return hash( 'sha256', $subject, false );
+	}
+
+	/**
+	 * Resets the delimiter to the pipe (|) character and creates an array of values.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @param mixed $value The delimiter separated meta value.
+	 *
+	 * @return string[]    String or array of meta values.
+	 */
+	private function maybe_explode( mixed $value ): array {
+		// Not digits
+		$pattern = '/(\D)/i';
+
+		// Reset the delimiter to pipe (|)
+		$value = preg_replace( $pattern, "|", $value );
+
+		// Explode the list of IDs into an array.
+		return explode( "|", $value );
+	}
+
+	/**
+	 * Update the meta field based on the found hash
+	 *
+	 * @since 1.0.0
+	 *
+	 * @param string $linked_post_type   The connected post type.
+	 * @param string $meta_key_to_update The meta field that needs to be updated.
+	 * @param int    $post_id            The ID of the last inserted post.
+	 * @param array  $record             The data of the last inserted post.
+	 * @param bool   $multiple           Whether we are importing more values for the same meta key.
+	 *
+	 * @return int|false                 Meta ID (add) or true (update) on success, false on failure.
+	 *
+	 */
+	private function update_linked_post_meta( string $linked_post_type, string $meta_key_to_update, int $post_id, array $record, bool $multiple = false ): bool|int {
+		$this->add_to_log( "Updating linked post meta..." );
+		$meta_key       = "_" . $linked_post_type . "_export_hash";
+		$this->meta_key = $meta_key;
+
+		$metafield_lowercase = strtolower( $meta_key_to_update );  // In the WPAI $record the meta keys come through as lowercase.
+
+		// Hash the old linked post type ID.
+		$meta_value = $this->hashit( $record[ $metafield_lowercase ] );
+		$this->meta_value = $meta_value;
+
+		// Grab the new post ID based on the hash.
+		$new_linked_post_id       = $this->grab_post_id_based_on_meta();
+		$this->new_linked_post_id = $new_linked_post_id;
+
+		// If there's an ID, update.
+		if ( $new_linked_post_id ) {
+			if ( $multiple ) {
+				return add_post_meta( $post_id, $meta_key_to_update, $new_linked_post_id );
+			} else {
+				return update_post_meta( $post_id, $meta_key_to_update, $new_linked_post_id );
+			}
+		}
+		return false;
+	}
+
+	/**
+	 * Retrieve post ID based on meta key = meta value pair.
+	 *
+	 * @param string $meta_key   The meta key.
+	 * @param string $meta_value The meta value.
+	 *
+	 * @return false|string|null
+	 */
+	function get_post_id_from_meta( string $meta_key, string $meta_value ): bool|string|null {
```

Comment:
This methods, and many others, do not all belong in this class.
This `Plugin` class should be the entry point, the controller delegating other, small and specialized objects, to execute specialized tasks.

My rule of thumb: if describing the responsibility of a class requires too much words, or "AND" and "OR", then it's likely too large and should be broken down.

```
Plugin: the add-on main controller; creates posts and the relationships between them, fixes incoherent states ...
```

vs

```
Plugin: the add-one entry point, coordinates the import operations.
Relationships: updates the relationships between imported posts.
Fixer: fixes some smaller import incoherencies after the import completed.
```
You get the idea.

Reply:
Actually not fully. I mean I get the concept, just don't quite know how to execute it. This is at least 1 level above my current knowledge. I'm keen to learn. This would take another "lesson with a beer" session.

Reply:
It's fine to leave the class as it is.
The same way you break down a method that is doing too many things into smaller methods that do less things, you would break down a large class that does many things to act as a coordinator of smaller classes that do, each, less things.

Reply:
Just to back Luca on this with another way of explaining it - @andrasguseo one way of looking at things is the classical "Object Oriented way" of defining classes. 

Think of a class as being a "thing" that has attributes (i.e. properties) and can do actions (i.e. methods). If the "thing" is doing/storing stuff that doesn't fit within the definition of the "thing", it should move to it's own separate definition. If you can't wrap your head around properties/methods that SHOULD and SHOULDN'T go in a class, the class likely isn't very well defined. It could become the classic `Main` class that ends up being 10000 lines, because how do you define what "main" does - the responsibility isn't well defined.

It's a bit of an art - don't let anyone tell you different. :)